From b4e03e62f082fe291622879b558f6471c88a6b4f Mon Sep 17 00:00:00 2001
From: Tom White <tom@cloudera.com>
Date: Wed, 12 Sep 2012 13:21:25 +0100
Subject: [PATCH 1194/1357] MR1: CLOUDERA-BUILD. JobClient constructors should call setConf(conf).

Reason: Bug
Ref: CDH-7900
Author: Tom White
(cherry picked from commit c166f79e00d08d182fc69010f1ef2a4b41f7f9cb)
---
 src/mapred/org/apache/hadoop/mapred/JobClient.java |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapred/JobClient.java b/src/mapred/org/apache/hadoop/mapred/JobClient.java
index 4f47276..2f73196 100644
--- a/src/mapred/org/apache/hadoop/mapred/JobClient.java
+++ b/src/mapred/org/apache/hadoop/mapred/JobClient.java
@@ -473,7 +473,6 @@ public class JobClient extends Configured implements MRConstants, Tool  {
    * @throws IOException
    */
   public JobClient(JobConf conf) throws IOException {
-    setConf(conf);
     init(conf);
   }
   
@@ -494,6 +493,7 @@ public class JobClient extends Configured implements MRConstants, Tool  {
    * @throws IOException
    */
   public void init(JobConf conf) throws IOException {
+    setConf(conf);
     String tracker = conf.get("mapred.job.tracker", "local");
     tasklogtimeout = conf.getInt(
       TASKLOG_PULL_TIMEOUT_KEY, DEFAULT_TASKLOG_TIMEOUT);
@@ -566,6 +566,7 @@ public class JobClient extends Configured implements MRConstants, Tool  {
    */
   public JobClient(InetSocketAddress jobTrackAddr, 
                    Configuration conf) throws IOException {
+    setConf(conf);
     this.ugi = UserGroupInformation.getCurrentUser();
     jobSubmitClient = createRPCProxy(jobTrackAddr, conf);
   }
-- 
1.7.0.4

